contracts/CourtRoom.sol

```diff
@@ -49,6 +56,22 @@ contract CaseContract is Owned {
 
 	}
 
+	function submitPromise(bytes32 _claimId,address beneficiary, uint256 blockNumber,
```

Comment:
on second thought payment validity should be handled by another contract with an ABI having only `valid` boolean. This could be a recurring subscription contract where the service provider can charge preauthorized service payments (see https://github.com/DecentLabs/ethereum-billcollector/tree/master/contracts) or a chequebook contract